mmdet/core/hook/yolox_lrupdater_hook.py

```diff
@@ -0,0 +1,70 @@
+from mmcv.runner import get_dist_info
+from mmcv.runner.hooks import HOOKS
+from mmcv.runner.hooks.lr_updater import (CosineAnnealingLrUpdaterHook,
+                                          annealing_cos)
+
+
+@HOOKS.register_module()
+class YOLOXLrUpdaterHook(CosineAnnealingLrUpdaterHook):
+    """YOLOX learning rate scheme.
+
+    There are two main differences between YOLOXLrUpdaterHook
+    and CosineAnnealingLrUpdaterHook.
+
+    1. When the current running epoch is greater than
+       `max_epoch-last_epoch`, a fixed learning rate will be used
+    2. The exp warmup scheme are different with LrUpdaterHook in MMCV
+
+    Args:
+        last_epoch (int): The number of last epoch with a fixed learning rate.
+        warmup_ratio (float): LR used at the beginning of warmup.
+           This parameter does not depend on the number of GPUs, so we need
+           to multiply by work_size.
+    """
+
+    def __init__(self, last_epoch, warmup_ratio, **kwargs):
+        _, work_size = get_dist_info()
+        self.base_lr_ = warmup_ratio * work_size
```

Comment:
Is the final underscore necessary?

Reply:
Because self.base_lr is already occupied internally.

Reply:
then leave a comment here to indicate the motivation